api/src/main/java/com/xwiki/googleapps/internal/GoogleAppsAuthService.java

```diff
@@ -19,17 +19,211 @@
  */
 package com.xwiki.googleapps.internal;
 
-import org.xwiki.component.annotation.Role;
+import java.net.URLEncoder;
+import java.security.Principal;
+import java.util.regex.Pattern;
 
-import com.xpn.xwiki.user.api.XWikiAuthService;
+import javax.servlet.http.HttpSession;
+
+import org.securityfilter.realm.SimplePrincipal;
+import org.slf4j.Logger;
+import org.xwiki.container.servlet.filters.SavedRequestManager;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.text.StringUtils;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.user.api.XWikiUser;
+import com.xpn.xwiki.user.impl.xwiki.XWikiAuthServiceImpl;
+import com.xpn.xwiki.web.XWikiRequest;
 
 /**
- * A badge interface to denote the role of the component that will replace the default authentication-service.
+ * An authenticator that can include a negotiation with the Google Cloud (e.g. Google Drive) services. This
+ * authenticator is created, configured and maintained by the GoogleAppsScriptService.
  *
  * @version $Id$
  * @since 3.0
  */
-@Role
-interface GoogleAppsAuthService extends XWikiAuthService
+public class GoogleAppsAuthService extends XWikiAuthServiceImpl
 {
+    private static final String XWIKISPACE = "XWiki.";
+
+    private final Logger log;
+
+    private final GoogleAppsXWikiObjects gaXwikiObjects;
+
+    private final CookieAuthenticationPersistence cookiePersistance;
+
+    private final Pattern logoutRequestMatcher;
+
+    GoogleAppsAuthService(GoogleAppsXWikiObjects gaXwikiObjects,
+            CookieAuthenticationPersistence cookiePersistance,
+            String logoutPattern, Logger log)
+    {
+        this.log = log;
+        this.gaXwikiObjects = gaXwikiObjects;
+        this.logoutRequestMatcher = Pattern.compile(logoutPattern);
+        this.cookiePersistance = cookiePersistance;
+    }
+
+    /**
+     * Evaluates if the user can be authenticated based on request info such as cookies.
+     *
+     * @param context the context representign the request
+     * @return a valid user, if found.
+     * @throws XWikiException if anything went wrong
+     */
+    public XWikiUser checkAuth(XWikiContext context) throws XWikiException
```

Comment:
Missing ``@Override``? Otherwise, it's hard to know if this is a new method or not. And the JavaDoc is normally not needed for an override.